pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -332,6 +334,15 @@ func (b *EndpointBuilder) BuildClusterLoadAssignment(endpointIndex *model.Endpoi
 	if svcPort == nil {
 		return buildEmptyClusterLoadAssignment(b.clusterName)
 	}
+
+	if features.EnableIngressWaypointRouting {
```

Comment:
please also check ambient enabled

Reply:
We do not do this for any other feature flag. Instead, the ambient feature flags are automatically set to `false` if ambient is not enabled, see `registerAmbient`